runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -222,11 +227,13 @@ private void processElementRecursively(final VertexHarness vertexHarness, final
     }
 
     // Recursively process all of the additional output elements.
-    vertexHarness.getAdditionalTagOutputChildren().keySet().forEach(tag -> {
-      while (!outputCollector.isEmpty(tag)) {
-        final Object element = outputCollector.remove(tag);
-        handleAdditionalOutputElement(vertexHarness, element, tag); // Recursion
-      }
+    vertexHarness.getIRVertex().getPropertyValue(AdditionalTagOutputProperty.class).ifPresent(tagToVertex -> {
+      tagToVertex.values().forEach(tag -> {
```

Comment:
Why is the `values()` of `tagToVertex` called `tag`?

Reply:
Thank you for pointing out. I may change the variable name to `dstVertexId`